src/common/host_device_vector.cu

```diff
@@ -46,8 +46,8 @@ template <typename T>
 struct HostDeviceVectorImpl {
   struct DeviceShard {
     DeviceShard()
-        : proper_size_(0), device_(-1), start_(0), perm_d_(false),
-        cached_size_(~0), vec_(nullptr) {}
+        : proper_size_{0}, device_{-1}, start_{0}, perm_d_{false},
+      cached_size_{0}, vec_{nullptr} {}
```

Comment:
cached_size_(~0) -> cached_size_{0}

Doesn't this change the behaviour due to removing the '~' operator ?